<launch>
    <!-- LiDAR sensor parameters -->
    <param name="scan_line" type="int" value="16" />
    <param name="minimum_range" type="double" value="0.3"/>

    <!-- Mapping parameters -->
    <param name="mapping_skip_frame" type="int" value="1" />
    <param name="mapping_line_resolution" type="double" value="0.2"/>
    <param name="mapping_plane_resolution" type="double" value="0.4"/>

    <!-- Map saving parameters -->
    <param name="save_map" type="bool" value="true"/>
    <param name="map_save_path" type="string" value="$(find aloam_velodyne)/maps/"/>
    <param name="map_file_name" type="string" value="aloam_map"/>

    <!-- A-LOAM nodes -->
    <node pkg="aloam_velodyne" type="ascanRegistration" name="ascanRegistration" output="screen">
        <remap from="/velodyne_points" to="/lidar_points"/>
    </node>

    <node pkg="aloam_velodyne" type="alaserOdometry" name="alaserOdometry" output="screen" />

    <node pkg="aloam_velodyne" type="alaserMapping" name="alaserMapping" output="screen" />

    <!-- RViz visualization -->
    <arg name="rviz" default="true" />
    <group if="$(arg rviz)">
        <node launch-prefix="nice" pkg="rviz" type="rviz" name="rviz" args="-d $(find aloam_velodyne)/rviz_cfg/aloam_velodyne.rviz" />
    </group>

</launch>